src/Normalizer/UnixMillisecondNormalizer.php

```diff
@@ -5,24 +5,65 @@
 use Lullabot\Mpx\DataService\DateTime\ConcreteDateTime;
 use Lullabot\Mpx\DataService\DateTime\ConcreteDateTimeInterface;
 use Lullabot\Mpx\DataService\DateTime\DateTimeFormatInterface;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
 use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
 use Symfony\Component\Serializer\Normalizer\DateTimeNormalizer;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 /**
  * Denormalize a millisecond timestamp into a date object.
+ *
+ * Note this doesn't extend the
+ * \Symfony\Component\Serializer\Normalizer\DateTimeNormalizer normalizer
+ * directly because the ::denormalize() method in that class is typehinted to
+ * \DateTimeInterface but we want to return a ConcreteDateTime object instead.
+ * Because of this, we decorate here a DateTimeNormalizer object and delegate
+ * all methods but ::denormalize().
  */
-class UnixMillisecondNormalizer extends DateTimeNormalizer
+class UnixMillisecondNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
 {
+    /**
+     * @var \Symfony\Component\Serializer\Normalizer\DateTimeNormalizer
+     */
+    private $decorated;
+
     private static array $supportedTypes = [
         DateTimeFormatInterface::class => true,
         ConcreteDateTimeInterface::class => true,
         ConcreteDateTime::class => true,
     ];
 
+    public function __construct(array $defaultContext = [])
+    {
+        $this->decorated = new DateTimeNormalizer($defaultContext);
+    }
+
+    public function setDefaultContext(array $defaultContext): void
+    {
+        $this->decorated->setDefaultContext($defaultContext);
+    }
+
+    public function getSupportedTypes(?string $format): array
+    {
+        return self::$supportedTypes;
```

Comment:
Ugh! If you don't override here it won't work due to _late static binding_, right? I wish the decorated used `static` instead of `self`.